CRM/Core/Form.php

```diff
@@ -2656,4 +2656,19 @@ public function setSelectedChild($default = NULL) {
     }
   }
 
+  /**
+   * Get the contact if from the url, using the checksum or the cid.
+   *
+   * @return string|bool
+   * @throws \CRM_Core_Exception
+   */
+  protected function getContactFromUrl(): bool {
+    $contactID = CRM_Utils_Request::retrieve('cid', 'Positive');
+    if (!$contactID) {
+      return FALSE;
+    }
+    $userChecksum = CRM_Utils_Request::retrieve('cs', 'String');
+    return (int) CRM_Contact_BAO_Contact_Utils::validChecksum($contactID, $userChecksum) ? $userChecksum : $contactID;
```

Comment:
So if the checksum doesn't validate we're still returning the contact id?

Reply:
Hmm - that might not be right - let me look again

Reply:
@seamuslee001 I think this is better